webui/react/src/e2e/models/hew/DataGrid.ts

```diff
@@ -0,0 +1,330 @@
+    const rows = await pwLocator.all();
+    return Promise.all(
+      rows.map(async (row) => {
+        return (
+          (await row.getAttribute(indexAttribute)) ||
+          Promise.reject(new Error(`all rows should have the attribute ${indexAttribute}`))
+        );
+      }),
+    );
+  }
+
+  /**
+   * Returns a list of rows that match the condition provided
+   * @param {(row: RowType) => Promise<boolean>} condition - function which tests each row against a condition
+   */
+  async filterRows(condition: (row: RowType) => Promise<boolean>): Promise<RowType[]> {
+    return (
+      await Promise.all(
+        Array.from(Array(await this.rows.pwLocator.count()).keys()).map(async (key) => {
+          const row = this.getRowByIndex(key);
+          return (await condition(row)) && row;
+        }),
+      )
+    ).filter((c): c is Awaited<RowType> => !!c);
+  }
+
+  /**
+   * Returns the row which matches the condition. Expects only one match.
+   * @param {string} columnName - column to read from
+   * @param {string} value - value to match
+   */
+  async getRowByColumnValue(columnName: string, value: string): Promise<RowType> {
+    const rows = await this.filterRows(async (row) => {
+      return (await row.getCellByColumnName(columnName).pwLocator.innerText()).indexOf(value) > -1;
+    });
+    if (rows.length !== 1) {
+      const names = await Promise.all(
+        rows.map(async (row) => await row.getCellByColumnName('Name').pwLocator.innerText()),
+      );
+      throw new Error(
+        `Expected one row to have ${columnName} ${value}. Found ${rows.length}: ${names}.`,
+      );
+    }
+    return rows[0];
+  }
+}
+
+/**
+ * Returns the representation of a Table Row.
+ * This constructor represents the Table in hew/src/kit/DataGrid.tsx.
+ * @param {object} obj
+ * @param {CanBeParent} obj.parent - The parent used to locate this Row
+ * @param {string} obj.selector - Used as a selector uesd to locate this object
+ * @param {DataGrid<RowType, HeadRowType>} [obj.parentTable] - Reference to the original table
+ */
+export class Row<
+  RowType extends Row<RowType, HeadRowType>,
+  HeadRowType extends HeadRow,
+> extends NamedComponent {
+  readonly defaultSelector = 'tr';
+  readonly indexAttribute = 'aria-rowindex';
+  constructor(args: RowArgs<RowType, HeadRowType>) {
+    super(args);
+    this.parentTable = args.parentTable;
+  }
+  parentTable: DataGrid<RowType, HeadRowType>;
+
+  /**
+   * Returns an aria label value for "selected" status
+   */
+  async isSelected(): Promise<string | null> {
+    return await this.pwLocator.getAttribute('aria-selected');
+  }
+
+  /**
+   * Returns the index of the row. Start counting at 0.
+   */
+  async getIndex(): Promise<number> {
+    const value = await this.pwLocator.getAttribute(this.indexAttribute);
+    if (value === null) {
+      throw new Error(`All rows should have the attribute ${this.indexAttribute}`);
+    }
+    return +value - 2;
```

Comment:
maybe check that value is a legit number here before casting with +. This seems like the type of thing that could cast an unexpected value without error.

Reply:
I can add it to line 190